### DISCUSSION TITLE
Firebase Kotlin SDK Tutorials

### DISCUSSION THREAD

**Discussion by mosabdullah** [GitLiveApp/firebase-kotlin-sdk#161] [Q&A]
Hi All,

I came across this initiative not long ago and think it is fantastic that we have a Kotlin firebase SDK. I know this is fairly new, but I was wondering if there are any tutorials on how to make use of the SDK in a multi-platform environment? 

I was looking for an authentication example, however, any example using any part of the SDK would do fine. I am new to the MPP and would like to further explore this option. 

Any help or pointers would be greatly appreciated. 

Thanks.

**Comment by nbransby** [comment]
Hi @mosabdullah im afraid we don't have anything official at the moment but we are looking to dedicate more resources to this project in the next few months and a tutorial will be on the top of the list

**Reply by mosabdullah** [reply]
Awesome, thanks.

**Reply by solvek** [reply]
Is there any example of a kotlin/js app which uses this sdk?

**Comment by carlosgub** [comment]
Hi @mosabdullah I create an article talking about implementing this library in Kotlin multiplatform https://medium.com/@carlosgub/how-to-implement-firebase-firestore-in-kotlin-multiplatform-mobile-with-compose-multiplatform-32b66cdba9f7

**Reply by akhilbalakrishnanl7** [reply]
Is there any example of integrating firebase-auth in Kotlin multiplatform?

**Reply by Thaerith** [reply]
Hello,

Except for firebase app initialisation, there are no major difference with "official" Firebase. So you can easily read official documentation about FirebaseAuth.

However, returned types can be different, which is a good thing in this case because it simplifies the use of some feature in asynchronous context.

e.g.

in official documentation, we have this:
``` kotlin
auth.createUserWithEmailAndPassword(email, password)
    .addOnCompleteListener(this) { task ->
        if (task.isSuccessful) {
            // Sign in success, update UI with the signed-in user's information
            Log.d(TAG, "createUserWithEmail:success")
            val user = auth.currentUser
            updateUI(user)
        } else {
            // If sign in fails, display a message to the user.
            Log.w(TAG, "createUserWithEmail:failure", task.exception)
            Toast.makeText(
                baseContext,
                "Authentication failed.",
                Toast.LENGTH_SHORT,
            ).show()
            updateUI(null)
        }
    }
```
    
with Firebase kotlin, `createUserWithEmailAndPassword` is a suspendable, so easy to use sequentially in a coroutine. you can even wrap it in a Flow...
    
``` kotlin
    try {
            val result = firebaseAuth.createUserWithEmailAndPassword(email, password)
            updateUI(result.user)
        } catch (e: Exception) {
            Toast.makeText(
                baseContext,
                "Authentication failed.",
                Toast.LENGTH_SHORT,
            ).show()
            updateUI(null)
        }
```

N.B.:  I didn't try this code. TBH, I usually not use the `AuthResult` from that function
